utils/forest_shim/src/machine/mod.rs

```diff
@@ -1,6 +1,22 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-pub use fvm::machine::Manifest as ManifestV2;
+use fvm::machine::MultiEngine as MultiEngine_v2;
+use fvm3::engine::MultiEngine as MultiEngine_v3;
 mod manifest_v3;
+pub use fvm::machine::Manifest as ManifestV2;
 pub use manifest_v3::ManifestV3;
+
+pub struct MultiEngine {
```

Comment:
Let's implement `Default` as well. It should take advantage of the available parallelism.